base/rational.jl

```diff
@@ -156,6 +156,9 @@ widen(::Type{Rational{T}}) where {T} = Rational{widen(T)}
 Approximate floating point number `x` as a [`Rational`](@ref) number with components
 of the given integer type. The result will differ from `x` by no more than `tol`.
 
+!!! note
+    `rationalize` does inexact conversion; for exact conversion use `convert` or the `Rational` constructor.
```

Comment:
```suggestion
    `rationalize` may perform inexact conversion even when `tol = 0`; for exact conversion use the [`Rational`](@ref) constructor.
```
I prefer `may perform` over `does` because `rationalize` may also perform exact conversion when possible `rationalize(.75) == .75`.

I would assume that `rationalize(x, tol=0)` performs exact conversion, but even this is not always the case. I think it is worth highlighting this especially (and let me know if you disagree, but I think it is pretty clear that if rationalize with tol=0 is inexact then rationalize with `tol > 0` will also be inexact).
```
julia> x = 1/big(7)
0.1428571428571428571428571428571428571428571428571428571428571428571428571428568

julia> rationalize(x, tol=0) == x
false

julia> iszero(rationalize(x, tol=0) - x)
true

julia> Rational(x) == x
true
```
Even though the difference is 0, the result is not equal to the input. 

As for recommendations, `convert(Rational, x)` is the same as `Rational(x)`, and I prefer to provide only one recommendation if the two options are identical. Because `Rational` is more concise and internally `convert` calls `Rational`, I prefer to recommend only `Rational`.

We should also link `Rational` to the appropriate documentation.